doc/whats_new/v1.2.rst

```diff
@@ -105,6 +115,10 @@ Changelog
   boolean. The type is maintained, instead of converting to `float64.`
   :pr:`25147` by :user:`Tim Head <betatim>`.
 
+- |API| :func:`utils.fixes.delayed` is deprecated in 1.2.1 and will be removed
```

Comment:
I think we need a changelog entry to introduce `utils.parallel.Parallel` as well.

Reply:
Rather than a new entry I think we can just expand this entry as suggested below to keep related changes together: